wandb/integration/yolov8/yolov8.py

```diff
@@ -0,0 +1,267 @@
+from typing import List, Optional
+
+from ultralytics.yolo.engine.model import YOLO
+from ultralytics.yolo.engine.trainer import BaseTrainer
+from ultralytics.yolo.utils import RANK
+from ultralytics.yolo.utils.torch_utils import get_flops, get_num_params
+from ultralytics.yolo.v8.classify.train import ClassificationTrainer
+
+import wandb
+from wandb.sdk.lib import telemetry
+
+
+class WandbCallback:
+    """
+    An internal YOLO model wrapper that tracks metrics, and logs models to Weights & Biases.
+    Usage:
+    ```python
+    from wandb.integration.yolov8.yolov8 import WandbCallback
+    model = YOLO("yolov8n.pt")
+    wandb_logger = WandbCallback(model,)
+    for event, callback_fn in wandb_logger.callbacks.items():
+        model.add_callback(event, callback_fn)
+    ```
+    """
+
+    def __init__(
+        self,
+        yolo: YOLO,
+        run_name: Optional[str] = None,
+        project: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        resume: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        A utility class to manage wandb run and various callbacks for the ultralytics YOLOv8 framework.
+        Args:
+            yolo: A YOLOv8 model that's inherited from `:class:ultralytics.yolo.engine.model.YOLO`
+            run_name: The name of the run to be created on Weights & Biases.
```

Comment:
could you add additional information about the naming convention you implemented here

Reply:
:+1: Updated the docstrings to be more clear on the defaults and types as well as the functionality of the arguments..